### ISSUE TITLE
ProcessSet#cleanup never gets called in >=6.5.4, process_count gets corrupted

### ISSUE THREAD

**Issue by toregeschliman** [mperham/sidekiq#5528]
Ruby version: 2.7.4 et al
Rails version: 6.0.5.1 et al
Sidekiq / Pro / Enterprise version(s): 6.5.4

We are seeing issues with several of our services when upgrading from 6.5.3 to 6.5.4 where retried jobs are run in a very uneven fashion. It looks like after https://github.com/mperham/sidekiq/commit/345ba002bc7035d8135d58f4c5dad6dadef298dc#diff-b9b984031dbbdd2bf7c8ac63eaaf0dabb0333ed71a565bac8e8aace8abcd0463L190 the `"processes"` set never gets cleaned up anymore unless the web UI or sidekiqmon is run (the only places `ProcessSet.new` remains). We are not entirely sure why we have processes that are failing to clean up after themselves on shutdown, but we have been able to confirm the problem and behavior with the following steps:

1. Observe a worker service (on 6.5.4) with very spiky retries and 43 processes running
2. Attached to the corresponding Redis instance:
```
> redis.scard("processes")
2406
```
3. Load the web UI attached to that Redis instance, go to "Busy" tab.
4. On same Redis instance
```
> redis.scard("processes")
43
```
5. Retry load becomes much more even.

Our suspicion is that as a bunch of ghost processes accrue, the `scaled_poll_interval` becomes much larger than it should be for the actual count of worker processes, so retries get re-queued in large and infrequent batches. We are trying to find the source of the process leak and will be observing to see how the count climbs over time, will update in the issue if we find anything meaningful.

**Comment by mperham** [comment]
The scheduler is supposed to have a 10% chance of cleaning the processes set when starting up. Did I get the logic wrong in schedule.rb? One of your 43 processes should run the cleanup…

**Comment by mperham** [comment]
Looks like It’s even more aggressive than that, every process tries to clean processes:

https://github.com/mperham/sidekiq/blob/d11c9c09f25a05db3d7bcf83a2e18d6b30007b51/lib/sidekiq/scheduled.rb#L233

but the first line of cleanup ensures it doesn’t happen more than once per minute. So what am I missing?

**Comment by toregeschliman** [comment]
Looks like that change isn't released yet; since 6.5.6 there's commits adding a new `ProcessSet.new` call and then replacing it to remove the Api dependency. (looking at https://github.com/mperham/sidekiq/compare/v6.5.6...main). Either way it looks like next patch release should address it fine.
